server/metrics/tracing.go

```diff
@@ -212,6 +208,9 @@ func (s *SpanMeta) StartTracing(ctx context.Context, childOnly bool) context.Con
 }
 
 func (s *SpanMeta) FinishTracing(ctx context.Context) context.Context {
+	if !s.started {
+		log.Error().Str("service_name", s.serviceName).Str("resource_name", s.resourceName).Msg("Finish tracing called before starting the trace")
```

Comment:
don't we need to return as well from here?

Reply:
I was thinking about this if we should return, panic, or just log in here. I ended up with logging to see how it goes. If this is called without a start tracing, it is a programming error.

Reply:
Added the return.